mm2src/coins/utxo/rpc_clients/connection_manager_multiple.rs

```diff
@@ -0,0 +1,409 @@
+use async_trait::async_trait;
+use core::time::Duration;
+use futures::lock::{Mutex as AsyncMutex, MutexGuard};
+use futures::{select, FutureExt};
+use std::collections::HashMap;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use crate::utxo::ScripthashNotificationSender;
+use common::executor::abortable_queue::{AbortableQueue, WeakSpawner};
+use common::executor::{AbortableSystem, SpawnFuture, Timer};
+use common::log::{debug, error, info, warn};
+
+use super::connection_manager_common::{ConnectionManagerErr, ConnectionManagerTrait, DEFAULT_CONN_TIMEOUT_SEC,
+                                       SUSPEND_TIMEOUT_INIT_SEC};
+use super::{spawn_electrum, ElectrumClientEvent};
+use super::{ElectrumConnSettings, ElectrumConnection};
+
+#[derive(Clone, Debug)]
+pub struct ConnectionManagerMultiple(pub Arc<ConnectionManagerMultipleImpl>);
+
+#[derive(Debug)]
+pub struct ConnectionManagerMultipleImpl {
```

Comment:
no, this doesn't work. same reason as above https://github.com/KomodoPlatform/komodo-defi-framework/pull/1966#discussion_r1489254384

Reply:
Ahaa, since we clone `self` in some methods, got you.

We could do `impl ConnectionManagerTrait for Arc<ConnectionManagerMultipleImpl>` and it actually makes more sense than `impl ConnectionManagerTrait for ConnectionManagerMultipleImpl` cause `ConnectionManagerMultipleImpl` can't live without the `Arc` anyways.

Reply:
not really `ConnectionManagerTrait` should be implemented for `ConnectionManagerMultipleImpl` because it's the structure we need to get access to the impls when trying to access using `trait object ` like here https://github.com/KomodoPlatform/komodo-defi-framework/blob/e6f2ba33722e8d4c03e56da769a0dbe3a37853b0/mm2src/coins/utxo/rpc_clients.rs#L1661C36-L1661C91